### DISCUSSION TITLE
Secure MySQL

### DISCUSSION THREAD

**Discussion by benjaminbowles** [filamentphp/filament#7188] [Help]
Anybody have any issues with secure mysql connections with PlanetScale.

Rest of the application and CLI is fine using MYSQL_ATTR_SSL_CA, but Filament resources give me the following error:

`SQLSTATE[HY000] [1105] unknown error: Code: UNAVAILABLE server does not allow insecure connections, client must use SSL/TLS`

**Comment by leandrocfe** [comment]
Did you try this?
```php
'mysql' => [
...
'options'   => array(
    PDO::MYSQL_ATTR_SSL_CA      => '/home/.../ca-cert.pem',
    PDO::MYSQL_ATTR_SSL_CERT    => '/home/.../cert.pem',
    PDO::MYSQL_ATTR_SSL_KEY     => '/home/.../key.pem'
),
'strict' => true,
'engine' => null,
],
```
https://stackoverflow.com/questions/53061182/mysql-connection-over-ssl-with-laravel

**Reply by benjaminbowles** [reply]
What is weird is that my main Laravel application and the CLI work fine, I can run `artisan db` and it connects to PlanetScale fine. It's just when I'm using Filament resources, I get these connection issues.

**Reply by leandrocfe** [reply]
are you running a local app?
Some problems here with relationships but my local app is running with PanetScale

![image](https://github.com/filamentphp/filament/assets/3833889/4a7cadf7-51f8-4757-9c0c-80cce222d598)

MYSQL_ATTR_SSL_CA=/etc/ssl/certs/ca-certificates.crt

**Reply by leandrocfe** [reply]
what about the path to the system CA?
https://planetscale.com/docs/concepts/secure-connections#ca-root-configuration

I'm using ubuntu
/etc/ssl/certs/ca-certificates.crt

**Reply by benjaminbowles** [reply]
Running on forge, my database.php config:

`'options' => [
 PDO::MYSQL_ATTR_SSL_CA =>  env('MYSQL_ATTR_SSL_CA'),
 PDO::MYSQL_ATTR_SSL_VERIFY_SERVER_CERT => false,
PDO::MYSQL_ATTR_SSL_CERT    => env('MYSQL_ATTR_SSL_CERT'),
:MYSQL_ATTR_SSL_KEY     => env('MYSQL_ATTR_SSL_KEY')
],`

Works perfectly for the laravel/CLI, just not on filament.

**Reply by leandrocfe** [reply]
take a look
https://planetscale.com/docs/concepts/secure-connections#ca-root-configuration

**Reply by benjaminbowles** [reply]
Think your missing my point here.

I am connected to PlanetScale perfectly fine using a ca-cert within my laravel application and the CLI, I can connect/interact with the PS database using a secure connection perfectly well.

Apart from when I use filament, when its not connecting to the database securely and I can't work out why.

**Reply by leandrocfe** [reply]
hum ok. It was supposed to work.
I said about the certificate because if I remove `/etc/ssl/certs/ca-certificates.crt`, I get the same error.

It is working here.
Maybe you can try it locally... If I can help you, let me know...
